src/reporting_function.tf

```diff
@@ -0,0 +1,139 @@
+resource "azurerm_resource_group" "reporting_fdr_rg" {
+  count = var.reporting_fdr_enabled ? 1 : 0
+
+  name     = format("%s-reporting-fdr-rg", local.project)
+  location = var.location
+
+  tags = var.tags
+}
+
+# Subnet to host reporting-fdr function
+module "reporting_fdr_function_snet" {
+  count                                          = var.reporting_fdr_enabled && var.cidr_subnet_reporting_fdr != null ? 1 : 0
```

Comment:
Is it really required this flag **reporting_fdr_enabled** ?

Reply:
no you're right i'll remove it